### PR TITLE
Cookiecutter

### PR DESCRIPTION
**Pull request by khiron**
OK this was a wholesale change of pretty much all files.

The cookiecutter prompts for 
- library_name: name of the library under the ./src folder
- apps: comma delimited list of apps for the library to expose
- author
- email
- project_slug: the name of the root directory for the project and also the likely repo name

All tests are setup with dummy data and marked as xfail'ing 

NB: I don't change the notebook `./docs.trial_plugins.ipynb` based on the cookiecutter fields.  Right now that's an exercise for the developer.  But I plan to automate that too when I get some time.

### CODE REVIEW

File: cookiecutter-cogent3-plugin/hooks/post_gen_project.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from jinja2 import Template
+
+# Load the template content
+app_template_content = """
+{% raw %}
+from cogent3.app.composable import define_app
+
+@define_app()
+class {{ app_class_name }}:
+    def main(self, data: str) -> str:
```
Comment (GavinHuttley) [inline cookiecutter-cogent3-plugin/hooks/post_gen_project.py:20]: If you're going to define a class, may as well add a basic `__init_` method to the stub too. Also, can you add a commented out version that just uses a function intsead of the class.

**Review by GavinHuttley — CHANGES_REQUESTED** [review]
one modification.

Also, have you checked its execution creates the directory structure you expect?

File: cookiecutter-cogent3-plugin/hooks/post_gen_project.py (hunk 1 above)
Comment (khiron) [reply cookiecutter-cogent3-plugin/hooks/post_gen_project.py:20]: will do (commented out function, and __init__).

I have been running (many many times) the cookiecutter and then `pip install .`  the resulting folder and looking at the notebook and seeing the apps.  BTW almost have the notebook working as a template ... maybe another 30 mins I'll have that working too.

I _think_ it's creating the directory structure I intend ... but I may be missing something obvious.

Comment (khiron) [reply cookiecutter-cogent3-plugin/hooks/post_gen_project.py:20]: I'm going to add some sample docstrings too, so people know how to document apps

**Pull request merged**